library/src/main/java/com/mux/player/internal/cache/CacheDatastore.kt

```diff
@@ -393,31 +369,33 @@ internal class CacheDatastore(
   // If the db failed to open, this method will throw. Opening can be re-attempted after resolving
   @Throws(IOException::class)
   private fun awaitDbHelper(): DbHelper {
-    fun closeIfInterrupted(dbHelper: DbHelper?) {
+    // inner function closes db if the calling thread was interrupted
+    fun closeAndCancelIfInterrupted(dbHelper: DbHelper?) {
       if (Thread.interrupted()) {
         dbHelper?.close()
         throw CancellationException("open interrupted")
       }
     }
 
+    // inner function prepares the database & cache dir + opens the DB
     fun doOpen(): DbHelper {
       // todo - we should also consider getting our cacheQuota here, that will take a long time
       //  so maybe do it async & only consider the cache quota once we have it(..?)
-      closeIfInterrupted(null)
+      closeAndCancelIfInterrupted(null)
       clearTempFiles()
-      closeIfInterrupted(null)
+      closeAndCancelIfInterrupted(null)
```

Comment:
Are these calls still necessary if they're passed `null`?

Reply:
Yep. They cancel the task that's opening the DB.

If anything, we need the `CancelationException` more than we do the `close` in this case.